app/browser-sync.ts

```diff
@@ -93,6 +95,16 @@ export class TargetLocatorSync {
     exec(this.targetLocator.window(nameOrHandle));
   }
 
+  frame(nameOrElement: ElementFinderSync | string): void {
```

Comment:
I'm not sure I understand. I am attempting to give flexibility by accepting `ElementFinderSync | string` so the caller can either pass the name of the frame (which we're currently doing in Ultra), or locate the element some other way and pass that. Isn't that very similar to what you're suggesting?

I suppose to be the _most_ flexible I could also accept a Locator and have the method look up the proper element that way.